gap/pairingheap.gi

```diff
@@ -101,19 +92,60 @@ function(heap)
         fi;
     end;
 
-    merge_pairs := function(isLess, heaps)
-        local h, res;
+    # TODO: Find out whether recursive vs non-recursive
+    #       has any measurable benefits
```

Comment:
Quick test suggest that the _nr version is slower: After tweaking `TestHeap` to use 10000 items again (instead of 3000), I get this from running our .tst files a few times each
```
binary heap: ca. 150 msec
pairing heap recursive: ca. 460 msec
pairing heap non-recursive: ca. 540 msec
```
So the _nr version seems clearly inferior, at least the one implemented below.